deltacat/io/memcached_object_store.py

```diff
@@ -19,25 +20,39 @@ class MemcachedObjectStore(IObjectStore):
     An implementation of object store that uses Memcached.
     """
 
-    def __init__(self, port=11212) -> None:
+    def __init__(
+        self, storage_node_ips: Optional[List[str]] = None, port: Optional[int] = 11212
+    ) -> None:
         self.client_cache = {}
         self.current_ip = None
         self.SEPARATOR = "_"
         self.port = port
+        self.storage_node_ips = storage_node_ips
+        self.hasher = None
         super().__init__()
 
+    def initialize_hasher(self):
+        if not self.hasher and self.storage_node_ips:
+            self.hasher = RendezvousHash()
+            for n in self.storage_node_ips:
+                self.hasher.add_node(n)
+
     def put_many(self, objects: List[object], *args, **kwargs) -> List[Any]:
         input = {}
         result = []
         current_ip = self._get_current_ip()
+        if self.storage_node_ips:
+            create_ref_ip = self._get_storage_node_ip(current_ip)
```

Comment:
Should we call get_node with the key instead of current ip to ensure the uniform distribution?